.github/workflows/CI.yml

```diff
@@ -0,0 +1,59 @@
+name: Build and Test k-NN
+on:
+  push:
+    branches:
+      - master
+      - opendistro-*
+
+jobs:
+  Build-k-NN:
+    strategy:
+      matrix:
+        java: [12]
+
+    name: Build and Test k-NN Plugin
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout k-NN
+        uses: actions/checkout@v1
+        
+      - name: Setup Java ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+      
+      - name: Run build
+        run: |
+          ./gradlew build
+          ls -ltr build/distributions/
+          
+      - name: Pull and Run Docker
+        run: |
+          plugin=`ls build/distributions/*.zip`
+          version=`echo $plugin|awk -F- '{print $3}'| cut -d. -f 1-3`
+          plugin_version=`echo $plugin|awk -F- '{print $3}'| cut -d. -f 1-4`
+          echo $version
+          cd ..
+          
+          if docker pull opendistroforelasticsearch/opendistroforelasticsearch:$version
+          then
+            echo "FROM opendistroforelasticsearch/opendistroforelasticsearch:$version" >> Dockerfile
+            echo "RUN if [ -d /usr/share/elasticsearch/plugins/opendistro_security ]; then /usr/share/elasticsearch/bin/elasticsearch-plugin remove opendistro_security; fi" >> Dockerfile
+            echo "RUN if [ -d /usr/share/elasticsearch/plugins/opendistro-knn ]; then /usr/share/elasticsearch/bin/elasticsearch-plugin remove opendistro-knn; fi" >> Dockerfile
+            echo "ADD k-NN/build/distributions/opendistro-knn-$plugin_version.zip /tmp/" >> Dockerfile
+            echo "RUN /usr/share/elasticsearch/bin/elasticsearch-plugin install --batch file:/tmp/opendistro-knn-$plugin_version.zip" >> Dockerfile
+            
+            docker build -t odfe-knn:test .
+          fi
+          
+      - name: Run Docker Image
+        run: |
+          cd ..
+          docker run -p 9200:9200 -d -p 9600:9600 -e "discovery.type=single-node" odfe-knn:test
+          sleep 15
```

Comment:
Does `sleep 15` ever produce errors when the cluster isn't ready after 15 seconds?

Reply:
No. This is for the cluster to initialize and get ready.

Reply:
Sorry, what I meant to ask is 15 second sleep always enough time?